TestAddNetworkPolicyWithMultipleRules unit test is flaky?

**Describe the bug**
This is the first time I observe this test failing, but here is the error log I got for my CPU resource requests PR (https://github.com/vmware-tanzu/antrea/pull/659), which should not impact any unit test whatsoever:
```
2020-04-28T17:14:39.8288600Z --- FAIL: TestAddNetworkPolicyWithMultipleRules (0.01s)
2020-04-28T17:14:39.8295810Z     networkpolicy_controller_test.go:402: 
2020-04-28T17:14:39.8296124Z         	Error Trace:	networkpolicy_controller_test.go:402
2020-04-28T17:14:39.8296291Z         	Error:      	Not equal: 
2020-04-28T17:14:39.8296917Z         	            	expected: &v1beta1.NetworkPolicy{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"policy1", GenerateName:"", Namespace:"ns1", SelfLink:"", UID:"policy1", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Rules:[]v1beta1.NetworkPolicyRule{v1beta1.NetworkPolicyRule{Direction:"In", From:v1beta1.NetworkPolicyPeer{AddressGroups:[]string{"addressGroup1"}, IPBlocks:[]v1beta1.IPBlock(nil)}, To:v1beta1.NetworkPolicyPeer{AddressGroups:[]string(nil), IPBlocks:[]v1beta1.IPBlock(nil)}, Services:[]v1beta1.Service{v1beta1.Service{Protocol:(*v1beta1.Protocol)(0xc0002f6a90), Port:(*intstr.IntOrString)(0xc0002bdcc0)}}}, v1beta1.NetworkPolicyRule{Direction:"Out", From:v1beta1.NetworkPolicyPeer{AddressGroups:[]string(nil), IPBlocks:[]v1beta1.IPBlock(nil)}, To:v1beta1.NetworkPolicyPeer{AddressGroups:[]string{"addressGroup2"}, IPBlocks:[]v1beta1.IPBlock(nil)}, Services:[]v1beta1.Service{v1beta1.Service{Protocol:(*v1beta1.Protocol)(0xc0002f6a90), Port:(*intstr.IntOrString)(0xc0002bdcc0)}}}}, AppliedToGroups:[]string{"appliedToGroup1"}}
2020-04-28T17:14:39.8297701Z         	            	actual  : &v1beta1.NetworkPolicy{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"policy1", GenerateName:"", Namespace:"ns1", SelfLink:"", UID:"policy1", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Rules:[]v1beta1.NetworkPolicyRule{v1beta1.NetworkPolicyRule{Direction:"Out", From:v1beta1.NetworkPolicyPeer{AddressGroups:[]string(nil), IPBlocks:[]v1beta1.IPBlock(nil)}, To:v1beta1.NetworkPolicyPeer{AddressGroups:[]string{"addressGroup2"}, IPBlocks:[]v1beta1.IPBlock(nil)}, Services:[]v1beta1.Service{v1beta1.Service{Protocol:(*v1beta1.Protocol)(0xc0002f6a90), Port:(*intstr.IntOrString)(0xc0002bdcc0)}}}, v1beta1.NetworkPolicyRule{Direction:"In", From:v1beta1.NetworkPolicyPeer{AddressGroups:[]string{"addressGroup1"}, IPBlocks:[]v1beta1.IPBlock(nil)}, To:v1beta1.NetworkPolicyPeer{AddressGroups:[]string(nil), IPBlocks:[]v1beta1.IPBlock(nil)}, Services:[]v1beta1.Service{v1beta1.Service{Protocol:(*v1beta1.Protocol)(0xc0002f6a90), Port:(*intstr.IntOrString)(0xc0002bdcc0)}}}}, AppliedToGroups:[]string{"appliedToGroup1"}}
2020-04-28T17:14:39.8298048Z         	            	
2020-04-28T17:14:39.8298188Z         	            	Diff:
2020-04-28T17:14:39.8298649Z         	            	--- Expected
2020-04-28T17:14:39.8298978Z         	            	+++ Actual
2020-04-28T17:14:39.8299319Z         	            	@@ -1,2 +1,2 @@
2020-04-28T17:14:39.8300273Z         	            	-(*v1beta1.NetworkPolicy)(&NetworkPolicy{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:policy1,GenerateName:,Namespace:ns1,SelfLink:,UID:policy1,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,ManagedFields:[],},Rules:[{In {[addressGroup1] []} {[] []} [{0xc0002f6a90 80}]} {Out {[] []} {[addressGroup2] []} [{0xc0002f6a90 80}]}],AppliedToGroups:[appliedToGroup1],})
2020-04-28T17:14:39.8301310Z         	            	+(*v1beta1.NetworkPolicy)(&NetworkPolicy{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:policy1,GenerateName:,Namespace:ns1,SelfLink:,UID:policy1,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,ManagedFields:[],},Rules:[{Out {[] []} {[addressGroup2] []} [{0xc0002f6a90 80}]} {In {[addressGroup1] []} {[] []} [{0xc0002f6a90 80}]}],AppliedToGroups:[appliedToGroup1],})
2020-04-28T17:14:39.8301531Z         	            	 
2020-04-28T17:14:39.8301698Z         	Test:       	TestAddNetworkPolicyWithMultipleRules
2020-04-28T17:14:39.8302116Z E0428 17:14:39.805878    9637 reconciler.go:409] Error releasing Openflow ID for ofRule 8: ID 8 was not allocated, can't be released
2020-04-28T17:14:39.8302600Z E0428 17:14:39.810752    9637 reconciler.go:409] Error releasing Openflow ID for ofRule 8: ID 8 was not allocated, can't be released
2020-04-28T17:14:39.8303048Z E0428 17:14:39.810832    9637 reconciler.go:409] Error releasing Openflow ID for ofRule 9: ID 9 was not allocated, can't be released
2020-04-28T17:14:39.8303182Z FAIL
```

I don't know how often this occur. May be worth running the test 1000 times in a row with `-count` to check :)

**Versions:**
Antrea version: top-of-tree v0.6.0-dev

--- a/pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go
+++ b/pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go
@@ -399,7 +399,8 @@ func TestAddNetworkPolicyWithMultipleRules(t *testing.T) {
 			t.Fatal("Expected two rule updates, got timeout")
 		}
 	}
-	assert.Equal(t, policy1, controller.GetNetworkPolicy(policy1.Name, policy1.Namespace))
+	assert.ElementsMatch(t, policy1.Rules, controller.GetNetworkPolicy(policy1.Name, policy1.Namespace).Rules)
+	assert.ElementsMatch(t, policy1.AppliedToGroups, controller.GetNetworkPolicy(policy1.Name, policy1.Namespace).AppliedToGroups)
 	assert.Equal(t, 1, controller.GetNetworkPolicyNum())
 	assert.Equal(t, 2, controller.GetAddressGroupNum())
 	assert.Equal(t, 1, controller.GetAppliedToGroupNum())

